test/WebJobs.Script.Tests.Integration/EndToEndTestFixture.cs

```diff
@@ -168,6 +177,46 @@ protected virtual void CreateTestStorageEntities()
             ServiceBusQueueClient = Microsoft.ServiceBus.Messaging.QueueClient.CreateFromConnectionString(connectionString, serviceBusQueueName);
         }
 
+        private IProxyClientGenerator GetMockProxyClientGenerator()
+        {
+            var proxyClientGenerator = new Mock<IProxyClientGenerator>();
+            var proxyClient = new Mock<IProxyClient>();
+
+            ProxyData proxyData = new ProxyData();
+            proxyData.Routes.Add(new Routes()
+            {
+                Id = 1001,
+                Method = HttpMethod.Get,
+                Name = "test",
+                UrlTemplate = "/myproxy"
+            });
+
+            proxyData.Routes.Add(new Routes()
+            {
+                Id = 1002,
+                Method = HttpMethod.Get,
+                Name = "localFunction",
+                UrlTemplate = "/mymockhttp"
+            });
+
+            proxyClient.Setup(p => p.GetProxyData()).Returns(proxyData);
+
+            proxyClient.Setup(p => p.CallAsync(It.IsAny<object[]>(), It.IsAny<IFuncExecutor>(), It.IsAny<ILogger>())).Returns(
+                (object[] arguments, IFuncExecutor funcExecutor, ILogger logger) =>
+                {
+                    object requestObj = arguments != null && arguments.Length > 0 ? arguments[0] : null;
+                    var request = requestObj as HttpRequestMessage;
+                    var response = new HttpResponseMessage(HttpStatusCode.OK);
+                    response.Headers.Add("myversion", "123");
+                    request.Properties[ScriptConstants.AzureFunctionsHttpResponseKey] = response;
+                    return Task.CompletedTask;
+                });
+
+            proxyClientGenerator.Setup(p => p.CreateProxyClient(It.IsAny<string>(), It.IsAny<ILogger>())).Returns(proxyClient.Object);
```

Comment:
>(It.IsAny<string>(), It.IsAny<ILogger>() [](start = 63, length = 40)

Can we test these parameters? IE, we want to test that ScriptHost is correctly passing in the JSON and the ILogger.